lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -790,7 +797,13 @@ public synchronized State getConnectionState() {
     public synchronized void connect() {
         /* connect() is the only action that will bring the ConnectionManager out of a terminal currentState */
         if(currentState.terminal || currentState.state == ConnectionState.initialized) {
-            startup();
+            try {
+                startup();
+            } catch(InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
```
     Thread.currentThread().interrupt();
```
Does this mean we are interrupting external thread

Reply:
Maybe we should rethrow exception and let user handle it externally wdyt

Reply:
It's just the way java works, we are restoring interruption flag, we don't interrupt any thread. We can't rethrow, we can wrap in some `RuntimeException`, but it probably won't be useful at all. If somebody interrupts caller thread when we are waiting termination, I believe it's expecting behavior to ignore call to `connect()`. It's extremely rare scenario anyway, it also can be harmful to throw unexpected runtime exception

Reply:
Okay, so we are following https://www.baeldung.com/java-interrupted-exception#2-restore-the-interrupt. 
This means we will restore the status of being interrupted and caller method will be responsible for handling it right